packages/ui/src/types/authenticator/validator.ts

```diff
@@ -1,23 +1,25 @@
+import { PasswordSettings } from '.';
 import { AuthFormData } from './form';
 import { AuthChallengeNames } from './user';
 
 /**
  * Maps each input to its validation error, if any
  */
-export type ValidationError = Record<string, string>;
+export type ValidationError = Record<string, string | string[]>;
```

Comment:
Sorry on another nit (on what I previously suggested!) -- what do you think about always typing this to `string[]` instead of `string | string[]`?  ie. enforce validators to always return an array of errors.

Then we don't have to do conditional `as string` or `as string[]` on wherever validation errors are, and we would just do a simple for loop like you did for password fields above.

Reply:
Think that allows for greater extensibility in the future, for example if we have multiple validation rules placed on a username field

Reply:
Wouldn't this be a breaking change though? We have a `validateCustomSignUp` service that customers can override to add in their own validation rules. To this point we have been assuming the customers sends over an object, with the key being the string. 

Like this

```services={{
        async validateCustomSignUp(formData) {
          if (!formData.acknowledgement) {
            return {
              acknowledgement: 'You must agree to the Terms & Conditions',
            };
          }
        },
      }}
```

If we change the result to be an array, then won't this break?  

It is kind of annoying have to use TypeScript assertion everywhere now. But not sure if this can be done.

Reply:
Oh yeah, you’re right about the breaking change. Let’s leave as-is now, and have it covered in our 3.0 release.

Reply:
```suggestion
// todo(breaking): enforce error to always be array
export type ValidationError = Record<string, string | string[]>;
```